pkg/kube/claims/claim_create.go

```diff
@@ -0,0 +1,185 @@
+package claims
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/kube"
+	"github.com/skupperproject/skupper/pkg/kube/resolver"
+)
+
+type ClaimOptions struct {
+	Name     string
+	Password []byte
+	Expiry   time.Duration
+	Uses     int
+}
+
+type SiteContext interface {
```

Comment:
Would it make sense to move this to pkg/kube/site/context.go?

Reply:
Maybe ultimately. At present I see it as establishing the interface to an external context that this claims package requires (allowing the implementation to be decoupled from it). Different packages may have slightly different interfaces that need to be fulfilled. I do expect that there will be something that satisfies all of them. For now though, my view is that this interface is part of the claims package.